Extension/tests/safari-converter/test-safari-converter.js

```diff
@@ -216,6 +216,21 @@ QUnit.test("Generichide rules", function(assert) {
     assert.equal("document", convertedRule.trigger["resource-type"][0]);
 });
 
+QUnit.test("Generic domain sensitive rules", function(assert) {
+    var ruleText = '~google.com##banner';
```

Comment:
You should check rules order.

For instance, register:
`
[
'~example.org##generic',
'##wide1',
'##specific',
'@@||example.org^$generichide'
]
`

And check the sort order